src/components/PupilViews/PupilLessonOverview/PupilLessonOverview.view.tsx

```diff
@@ -1,67 +1,201 @@
+import { isString } from "lodash";
+
 import {
   OakBox,
+  OakBulletList,
   OakFlex,
+  OakGrid,
+  OakGridArea,
   OakHeading,
-  OakLI,
+  OakLessonBottomNav,
+  OakLessonLayout,
+  OakLessonNavItem,
   OakPrimaryButton,
-  OakSecondaryButton,
-  OakUL,
+  OakSpan,
+  OakSubjectIcon,
+  OakTertiaryButton,
+  isValidIconName,
 } from "@oaknational/oak-components";
-
 import {
-  lessonSections,
+  LessonSection,
   useLessonEngineContext,
 } from "@/components/PupilComponents/LessonEngineProvider";
 
-export const PupilViewsLessonOverview = () => {
+type PupilViewsLessonOverviewProps = {
+  lessonTitle: string;
+  yearTitle?: string;
+  subjectTitle: string;
+  subjectSlug: string;
+  pupilLessonOutcome?: string;
+  starterQuizNumQuestions: number;
+  exitQuizNumQuestions: number;
+};
+
+export const PupilViewsLessonOverview = ({
+  lessonTitle,
+  subjectTitle,
+  yearTitle,
+  subjectSlug,
+  pupilLessonOutcome,
+  exitQuizNumQuestions,
+  starterQuizNumQuestions,
+}: PupilViewsLessonOverviewProps) => {
   const {
     completedSections,
     sectionResults,
     updateCurrentSection,
     proceedToNextSection,
   } = useLessonEngineContext();
+  const subjectIconName: `subject-${string}` = `subject-${subjectSlug}`;
```

Comment:
This is really neat and new to me. The template literal type here intersects with the narrowing of `isValidIconName(subjectIconName)` on LN121 to form a type that fits into `OakSubjectIcon['iconName']` — that is any valid icon with a `subject-` prefix.

Reply:
didn't realise you could define types this way either